### DISCUSSION TITLE
Feature Request: Additional Branch list dropdown while manually triggering github actions workflow

### DISCUSSION THREAD

**Discussion by itsankit-google** [community/community#36383] [Actions]
While manually triggering the workflow, github actions provides a dropdown to use the workflow from any other branch in the repo.
I want to trigger the workflow from different branch and checkout different branch inside the workflow. Is it a way to provide an additional branch list dropdown using github inputs?? I know there is a `choice` option but that list will then be static and will not work any newly added branch in the repo.

I am able to workaround it right now using a string input but would be helpful if I can replicate a similar dropdown since it is already available to choose to use workflow from other branch.

![image](https://user-images.githubusercontent.com/88528384/196128488-51bac2aa-e374-418b-9c29-862866ed3ce4.png)

**Comment by jeroenhabets** [comment]
Sorry, to piggy back on this feature request but for one of our deploy workflows we'd love to have a **Tag list dropdown** while manually triggering a github actions worklow.

Our build action would create a tag. Our admin could select this tag in the deploy action to one of our environments. 

Also found as separate discussion in another part of this org: https://github.com/orgs/community/discussions/41138

**Comment by MMirabito** [comment]
Adding another use case in support of this request.

For our deployment workflows, we intentionally want the workflow definition to always run from `master`, while allowing the user to select a different branch as the source code to build and deploy.

The existing Use workflow from selector does not quite solve this because it couples the selected branch with the version of the workflow being executed. We want those two concerns to remain separate:

Workflow definition:  master
Source to deploy:   `  [ current repository branches ▼ ]`

Today we handle the source branch with a `workflow_dispatch `choice:

```
source_branch:
  description: 'Select source branch'
  required: true
  type: choice
  options:
    - master
    - Development_Branch
    - branch-1
    - branch-2
    - branch-3
```

This works, but it means maintaining a second list of branches inside the workflow. When a branch is created or deleted, the workflow also needs to be updated.

It would be useful if `workflow_dispatch` supported a repository-backed branch input, for example:

```
source_branch:
  description: 'Select source branch'
  required: true
  type: branch
```

GitHub could then populate the dropdown from the repository's current branches, similar to the existing branch selector.

The selected value could simply be consumed as a normal input:

- uses: actions/checkout@v4
  with:
    ref: ${{ inputs.source_branch }}

This would let us keep deployment logic controlled on `master `while dynamically selecting the application branch to deploy, without maintaining the same branch information in two places.


This is how we have implemented it so far, but it is not ideal. We also added a free-form input field that allows the developer to specify another branch at runtime if it is not available in the dropdown.

It works as a workaround, but having the dropdown dynamically populated with the repository's current branches would be a much cleaner solution.

<img width="383" height="597" alt="image" src="https://github.com/user-attachments/assets/9bbf1c2d-31c8-405a-8bce-a3cc16236f1b" />

A similar feature is also referenced here  [#41138](https://github.com/orgs/community/discussions/41138) and here [#44799](https://github.com/orgs/community/discussions/44799)

Thanks,
max
